### DISCUSSION TITLE
sleepwatcher not activating on wake or sleep

### DISCUSSION THREAD

**Discussion by mikespharss** [Homebrew/discussions#4231] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.21-99-g2a417f1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2a417f14163aacfb4ffcb754fe9dea71c46f6ad0
Last commit: 4 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9e5a23de71f24a17ee1b0652d3331f41fe639aa0
Core tap last commit: 3 weeks ago
Core tap branch: master
Core tap JSON: 17 Feb 16:14 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: /usr/bin/vim
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.38.1 => /usr/local/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  kubernetes-cli

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

### Description of issue

I installed sleepwatcher recently, created the .sleep and .wake scripts, but it doesn't seem to be triggering at all on sleep or wake. I've tried adding some logging output to those scripts to write to a file when they run, and the log file doesn't get updated when I force the laptop to sleep. They work when I invoke them directly though.

One thing that looked suspicious to me is the command-line options on the process started by homebrew service

```
  501 85694     1   0 Wed05PM ??         0:34.47 /usr/local/opt/sleepwatcher/sbin/sleepwatcher -V -s /Users/brew/actions-runner/_work/homebrew-core/homebrew-core/bottles/home/.sleep -w /Users/brew/actions-runner/_work/homebrew-core/homebrew-core/bottles/home/.wakeup
```

I googled around but didn't find any evidence that this is some sort of normal disambiguation of home directories in the homebrew environment, but those brew/actions-runner directories don't exist locally at all. I don't even see a `/Users/brew`

Am I barking up the wrong tree? Is that normal for home directories to be disambiguated to `/Users/brew/actions-runner/_work/homebrew-core/homebrew-core/bottles/home`?

**Comment by mikespharss** [comment]
Fwiw I just tested this by forcibly creating sleep and wake scripts at `/Users/brew/actions-runner/_work/homebrew-core/homebrew-core/bottles/home/` owned by me, and now the sleepwatcher scripts are being triggered on sleep and wake as expected.

It appears something about the homebrew build or install of sleepwatcher service is incorrectly setting the current user's home directory. Some remnant of github actions perhaps?

Just speculating but I wonder if it had something to do with this change: https://github.com/Homebrew/homebrew-core/commit/1676fcbb9c5b66cf41466b3e26993a0e1a0b49bb

**Reply by gromgit** [reply]
You got bit by an API install issue. See https://github.com/Homebrew/homebrew-core/issues/123613#issuecomment-1439401559 for the current workaround.

**Reply by mikespharss** [reply]
Aha, thank you! That's exactly what I was looking for.

**Comment by mikespharss** [comment, marked as answer]
For posterity, and to save someone a click later, the suggestion in the above link worked:

```
$ HOMEBREW_NO_INSTALL_FROM_API=1 brew reinstall sleepwatcher

$ brew services restart sleepwatcher
 ```

The running process looks correct now:
```
 /usr/local/opt/sleepwatcher/sbin/sleepwatcher -V -s /Users/msphar/.sleep -w /Users/msphar/.wakeup
```
